### DISCUSSION TITLE
gunicorn between fast cgi and django

### DISCUSSION THREAD

**Discussion by cederom** [benoitc/gunicorn#3177] [Q&A]
Hello world :-)

My shared hosting provider decided to remove apache mod_wsgi in a few days with only fast cgi working.

I have several django based websites that requires wsgi to work (support for fast cgi was removed long ago).

As gunicorn seems most popular wsgi server right now I am wondering if it can be configured to as wsgi gateway between apache fast cgi and django (i.e. not in a standalone long running process mode just a mod_wsgi replacement)?

I will have to either make gunicorn work as this kind of interface or change hosting provider and/or setup my own server.

I saw lots of questions like this over the years on the web but no clear answer. If gunicorn could do that probably more folks would be interested :-)

I was happy (for almost 15 years) with my current shared hosting (Linux with shell access) because at good price I could have several small web sites running on different technologies. But with time they removed good functionalities and now they force me to switch to 3x more expensive VPS that also adds server maintenance on top. That sounds like a good business model to pay someone more just to have more work yourself right?

Any hints welcome :-)

**Comment by benoitc** [comment]
In theory it should be feasible to launch it as a fastcgi process, but for now it can't be launched this way. I assume you need such feature because you can't launch it gunicorn as a standalone service  with your current hosting provider?

**Reply by cederom** [reply]
Yes exactly, no persistent process is allowed on shared hosting, persistent process is only available on VPS (+proxy), and I can see up to this day there is no interface to run Django on FastCGI not to say FastCGI-To-WSGI interface :-)

**Reply by cederom** [reply]
I am considering moving to Always-Free OCI instance (ARM64 4xCPU 24GB RAM 100GB storage) and setup my own FreeBSD server as alternative but there are so many websites / services running already and probably more people are in this situation :-)

**Reply by cederom** [reply]
Here is a reference discissuion from the Django Project Forum:
https://forum.djangoproject.com/t/fastcgi-django/28323/7

**Comment by jeremyherbert** [comment]
I have a similar problem. Also because of dreamhost.

**Reply by cederom** [reply]
Thanks @jeremyherbert !! We made it with a FastCGI + uWSGI + Django :-)
